relay-sampling/src/lib.rs

```diff
@@ -324,12 +341,64 @@ pub struct SamplingRule {
     #[serde(rename = "type")]
     pub ty: RuleType,
     pub id: RuleId,
+    /// The time range the rule should be applicable in.
+    ///
+    /// If no time range is provided, relay will always apply the rule. If a range is
+    /// provided the rule becomes a decayng rule, and relay will only apply it inside
+    /// the time ranges.
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub time_range: Option<TimeRange>,
 }
 
 impl SamplingRule {
     fn supported(&self) -> bool {
         self.condition.supported()
     }
+
+    /// Returns whether the sampling rule is a decaying rule.
+    ///
+    /// A decaying rule is a sampling rule that only applies in a closed time range.
+    fn is_decaying_rule(&self) -> bool {
+        match self.time_range {
+            None => false,
+            Some(_) => true,
+        }
+    }
+
+    /// Returns whether the sampling rule is active.
+    ///
+    /// Non-decaying rules are always active. Decaying rules are active if they are
+    /// neither idle nor expired.
+    fn is_active(&self) -> bool {
+        if !self.is_decaying_rule() {
+            return true;
+        }
+        if !self.is_idle() && !self.is_expired() {
+            return true;
+        }
+        false
+    }
+
+    /// Returns whether the sampling rule is an idle decaying rule.
+    ///
+    /// An idle decaying rule is a decaying rule that is not operative yet, but it will
+    /// in the future.
+    fn is_idle(&self) -> bool {
+        match self.time_range.as_ref() {
+            None => false,
+            Some(time_range) => !time_range.has_started(),
+        }
+    }
```

Comment:
It seems a little bit too verbose. Maybe something like this, in functional style, would be better for readability? 
```suggestion
    fn is_idle(&self) -> bool {
        !self
            .time_range
            .as_ref()
            .map_or(true, TimeRange::has_started)
    }
```